src/oscar/apps/dashboard/catalogue/forms.py

```diff
@@ -24,12 +24,22 @@
                                                   ('RelatedFieldWidgetWrapper',
                                                    'RelatedMultipleFieldWidgetWrapper'))
 
-CategoryForm = movenodeform_factory(
+
+BaseCategoryForm = movenodeform_factory(
```

Comment:
From the documentation at https://django-treebeard.readthedocs.io/en/latest/forms.html#treebeard.forms.movenodeform_factory
```
MyNodeForm = movenodeform_factory(MyNode)
```
is equivalent to
```
class MyNodeForm(MoveNodeForm):
    class Meta:
        model = models.MyNode
        exclude = ('sib_order', 'parent')
```

Wouldn't it be mode clear to define CategoryForm by subclassing MoveNodeForm instead of generating it with the factory?

Reply:
There's a warning right above the section in the treebeard documentation you link to: 

> Subclassing `MoveNodeForm` directly is discouraged, since special care is needed to handle excluded fields, and these change depending on the tree type. It is recommended that the `movenodeform_factory()` function is used instead.

Obviously we only use one tree type in Oscar but I think relying on the factory is more resilient to any future upstream changes?